### PR TITLE
More readable keybinding from lua example. #285

### PR DESCRIPTION
**Pull request by ehllie**
Addresses the issue mentioned in #285

### CODE REVIEW

**Comment by kittywitch** [comment]
See: https://github.com/phaazon/hop.nvim/issues/285#issuecomment-1186634318

Otherwise, seemingly doesn't work under NVIM v0.7.2.

**Review by kittywitch — CHANGES_REQUESTED** [review]
```
vim.keymap.set("", "f", function()
  hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true })
end, {remap=true})
vim.keymap.set("", "F", function()
  hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true })
end, {remap=true})
vim.keymap.set("", "t", function()
  hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true, hint_offset = -1 })
end, {remap=true})
vim.keymap.set("", "T", function()
  hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true, hint_offset = 1 })
end, {remap=true})
```

File: README.md [hunk 1]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
 
 ```lua
 -- place this in one of your configuration file(s)
-vim.api.nvim_set_keymap('', 'f', "<cmd>lua require'hop'.hint_char1({ direction = require'hop.hint'.HintDirection.AFTER_CURSOR, current_line_only = true })<cr>", {})
-vim.api.nvim_set_keymap('', 'F', "<cmd>lua require'hop'.hint_char1({ direction = require'hop.hint'.HintDirection.BEFORE_CURSOR, current_line_only = true })<cr>", {})
-vim.api.nvim_set_keymap('', 't', "<cmd>lua require'hop'.hint_char1({ direction = require'hop.hint'.HintDirection.AFTER_CURSOR, current_line_only = true, hint_offset = -1 })<cr>", {})
-vim.api.nvim_set_keymap('', 'T', "<cmd>lua require'hop'.hint_char1({ direction = require'hop.hint'.HintDirection.BEFORE_CURSOR, current_line_only = true, hint_offset = 1 })<cr>", {})
+local hop = require("hop")
+local directions = require("hop.hint").HintDirection
+vim.api.nvim_set_keymap("", "f", function()
```
Comment (kittywitch) [inline README.md:152]:
```suggestion
vim.keymap.set("", "f", function()
```

File: README.md [hunk 2]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+  hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true })
+end, {})
```
Comment (kittywitch) [inline README.md:154]:
```suggestion
end, {remap=true})
```

File: README.md [hunk 3]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+vim.api.nvim_set_keymap("", "F", function()
```
Comment (kittywitch) [inline README.md:155]:
```suggestion
vim.keymap.set("", "F", function()
```

File: README.md [hunk 4]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+  hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true })
+end, {})
```
Comment (kittywitch) [inline README.md:157]:
```suggestion
end, {remap=true})
```

File: README.md [hunk 5]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+vim.api.nvim_set_keymap("", "t", function()
```
Comment (kittywitch) [inline README.md:158]:
```suggestion
vim.keymap.set("", "t", function()
```

File: README.md [hunk 6]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+  hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true, hint_offset = -1 })
+end, {})
```
Comment (kittywitch) [inline README.md:160]:
```suggestion
end, {remap=true})
```

File: README.md [hunk 7]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+vim.api.nvim_set_keymap("", "T", function()
```
Comment (kittywitch) [inline README.md:161]:
```suggestion
vim.keymap.set("", "T", function()
```

File: README.md [hunk 8]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+  hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true, hint_offset = 1 })
+end, {})
```
Comment (kittywitch) [inline README.md:163]:
```suggestion
end, {remap=true})
```

**Review by kittywitch — CHANGES_REQUESTED** [review]
This should actually be a functional change request now. Sorry for the previous comment.

**Comment by ehllie** [comment]
> See: [#285 (comment)](https://github.com/phaazon/hop.nvim/issues/285#issuecomment-1186634318)
> 
> Otherwise, seemingly doesn't work under NVIM v0.7.2.
> 
> Comparatively:
> 
> ```lua
> vim.keymap.set("", "f", function()
>   hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true })
> end, {remap=true})
> vim.keymap.set("", "F", function()
>   hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true })
> end, {remap=true})
> vim.keymap.set("", "t", function()
>   hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true, hint_offset = -1 })
> end, {remap=true})
> vim.keymap.set("", "T", function()
>   hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true, hint_offset = 1 })
> end, {remap=true})
> ```
> 
> works fine.
> 
> In fact, https://github.com/ehllie/my-dotfiles/blob/main/.config/nvim/lua/user/hop.lua#L28-L32 your own dotfiles demonstrate this and not the function you provided.

Thanks for pointing out the error. I have gotten tripped up by the `vim.keymap.set` and `vim.api.nvim_set_keymap` difference before but seemed to have forgotten about it here.

**Comment by phaazon** [comment]
Eh, I like that `vim.keymap` thing.

File: README.md [hunk 9]
```diff
@@ -147,10 +147,20 @@ If you want to create a key binding from within Lua:
...
+end, {remap=true})
+vim.keymap.set("", "F", function()
+  hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true })
+end, {remap=true})
+vim.keymap.set("", "t", function()
+  hop.hint_char1({ direction = directions.AFTER_CURSOR, current_line_only = true, hint_offset = -1 })
+end, {remap=true})
+vim.keymap.set("", "T", function()
+  hop.hint_char1({ direction = directions.BEFORE_CURSOR, current_line_only = true, hint_offset = 1 })
+end, {remap=true})
```
Comment (phaazon) [inline README.md:163]: Can you follow the rest of the convention and use `'` and not `"` (unless you need to write `'` inside the string), and `require'hop'` not `require("hop")`. I agree that the later should be the norm but unfortunately people are more used to the former :disappointed:

**Review by phaazon — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (ehllie) [reply README.md:163]: I was just trying to change as little from the original. I don't really have string feelings either way, but I agree it should at least be consistent within projects

Comment (ehllie) [reply README.md]: I did not understand the message at the time, and though you had applied the changes yourself already. Pushed an update to change double quotes to single one.

**Review by phaazon — APPROVED** [review]

**Pull request merged**